core/src/main/java/jenkins/security/ClassFilterImpl.java

```diff
@@ -155,10 +157,25 @@ public boolean isBlacklisted(Class _c) {
                 LOGGER.log(Level.FINE, "permitting {0} since it is a throwable", name);
                 return false;
             }
-            if (Enum.class.isAssignableFrom(c)) { // Class.isEnum seems to be false for, e.g., java.util.concurrent.TimeUnit$6
```

Comment:
This comment describes exactly why I am using the construction above rather than `Class.isEnum`, for exactly the case of `TimeUnit` you bring up, so I do not understand what the purpose of this patch is.